File: Bystriy_start/#10_Eshe_odna_zadacha_na_teoriu_chisel/main.cpp
#include <iostream>

using ll = long long;

ll NOD(ll a, ll b)
{
    while (b != 0)
    {
        ll temp = a % b;
        a = b;
        b = temp;
    }
    return a;
}

ll NOK(ll a, ll b)
{
    return a * b / NOD(a, b);
}

int main()
{
    ll a, b;
    std::cin >> a >> b;

    std::cout << NOD(a, b) << " " << NOK(a, b) << "\n";

    return 0;
}

File: Bystriy_start/#10_Eshe_odna_zadacha_na_teoriu_chisel/main.py
def NOD(a, b):
    while b:
        b, a = a % b, b
    return a


def NOK(a, b):
    return a * b // NOD(a, b)


a, b = map(int, input().split())
print(NOD(a, b), NOK(a, b), sep=" ")
